pkg/controllers/disruption/controller.go

```diff
@@ -194,10 +203,42 @@ func (c *Controller) executeCommand(ctx context.Context, m Method, cmd Command,
 		return multierr.Append(fmt.Errorf("tainting nodes (command-id: %s), %w", commandID, err), state.RequireNoScheduleTaint(ctx, c.kubeClient, false, stateNodes...))
 	}
 
+	select {
+	case <-ctx.Done():
+		return fmt.Errorf("interrupted (command-id: %s)", commandID)
+	case <-c.clock.After(commandValidationDelay):
+	}
+
+	// Revalidate the proposed command after the node has been tainted.
+	// This addresses a race conditions where pods can schedule to the candidate nodes after the command has been finalized, but before the candidates were tainted.
+	candidates, err := GetCandidates(ctx, c.cluster, c.kubeClient, c.recorder, c.clock, c.cloudProvider, m.ShouldDisrupt, c.queue)
+	if err != nil {
+		return fmt.Errorf("validating command, determining candidates, %w", err)
+	}
+	disruptionBudgetMapping, err := BuildDisruptionBudgets(ctx, c.cluster, c.clock, c.kubeClient, c.recorder)
+	if err != nil {
+		return fmt.Errorf("validating command, building disruption budgets, %w", err)
+	}
+	validatedCommand := m.ValidateCommand(ctx, cmd, disruptionBudgetMapping, candidates...)
```

Comment:
From looking at the various `ValidateCommand()` calls, it looks like they all implement the following 

1. Map L214 candidates to command.Candidates
2. do `method.ShouldDisrupt`
3. check disruption budgets

I personally don't think we need another interface method for this, as the method specific behavior is already encapsulated in the GetCandidates() call on L214. I could see doing a helper function for the disruption budget mapping check. WDYT?

Reply:
Resolving, current approach removes the interface function. Realistically we only see this race occur for consolidation where the race window is expanded due to the validating scheduling simulation so I've added a second validation pass after that check and am going to opt to leave it at that for now.